skimage/restoration/tests/test_denoise.py

```diff
@@ -289,77 +289,92 @@ def test_denoise_bilateral_nan():
     assert_equal(img, out)
 
 
-def test_denoise_nl_means_2d():
+@pytest.mark.parametrize('fast_mode', [False, True])
+def test_denoise_nl_means_2d(fast_mode):
     img = np.zeros((40, 40))
     img[10:-10, 10:-10] = 1.
     sigma = 0.3
     img += sigma * np.random.randn(*img.shape)
+    img_f32 = img.astype('float32')
     for s in [sigma, 0]:
-        denoised = restoration.denoise_nl_means(img, 7, 5, 0.2, fast_mode=True,
-                                                multichannel=True, sigma=s)
-        # make sure noise is reduced
-        assert_(img.std() > denoised.std())
         denoised = restoration.denoise_nl_means(img, 7, 5, 0.2,
-                                                fast_mode=False,
-                                                multichannel=True, sigma=s)
+                                                fast_mode=fast_mode,
+                                                multichannel=True,
+                                                sigma=s)
         # make sure noise is reduced
         assert_(img.std() > denoised.std())
 
+        denoised_f32 = restoration.denoise_nl_means(img_f32, 7, 5, 0.2,
+                                                    fast_mode=fast_mode,
+                                                    multichannel=True,
+                                                    sigma=s)
+        # make sure noise is reduced
+        assert_(img.std() > denoised_f32.std())
+
+        # Sheck single precision result
+        assert np.allclose(denoised_f32, denoised, atol=1e-2)
```

Comment:
probably worth checking that the dtype matches our expectation

Reply:
actually I see you already added a separate test case for this below, so disregard this suggestion